observ/object_utils.py

```diff
@@ -0,0 +1,24 @@
+from functools import cache
+from itertools import chain
+
+
+@cache
+def get_class_slots(cls):
+    """utility to collect all __slots__ entries for a given type and its supertypes"""
+    # collect via iterables for performance
+    # deduplicate via set
+    return set(
+        chain.from_iterable(getattr(cls, "__slots__", []) for cls in cls.__mro__)
```

Comment:
Cool trick!

Reply:
`get_object_attrs` is really the silver bullet of this PR that makes it all possible